allennlp/data/tokenizers/token.py

```diff
@@ -40,16 +40,57 @@ class Token:
         added, similar to spacy's `lex_id`.
     """
 
-    text: Optional[str] = None
-    idx: Optional[int] = None
-    idx_end: Optional[int] = None
-    lemma_: Optional[str] = None
-    pos_: Optional[str] = None
-    tag_: Optional[str] = None
-    dep_: Optional[str] = None
-    ent_type_: Optional[str] = None
-    text_id: Optional[int] = None
-    type_id: Optional[int] = None
+    __slots__ = [
```

Comment:
Do you know if this impacts extensions to this class?  E.g., it would be sad to break this functionality: https://github.com/allenai/allennlp/pull/3769.

I just tried to answer this myself, and it looks like we won't break that functionality, which is good: https://stackoverflow.com/questions/1816483/how-does-inheritance-of-slots-in-subclasses-actually-work.

Reply:
It might be good to have a test for this